crates/amaru/tests/evaluate_ledger_states.rs

```diff
@@ -0,0 +1,223 @@
+    MintedWitnessSet, Point, PoolId, TransactionPointer, cbor, network::NetworkName,
+    protocol_parameters::ProtocolParameters,
+};
+use amaru_ledger::{
+    self,
+    context::DefaultValidationContext,
+    rules::transaction,
+    state::{self, volatile_db::StoreUpdate},
+    store::{EpochTransitionProgress, ReadStore, Store, TransactionalContext},
+};
+use amaru_stores::rocksdb::RocksDB;
+use pallas_addresses::Network;
+use pallas_codec::utils::AnyCbor;
+use std::{collections::BTreeSet, env, fs, iter, ops::Deref, path::PathBuf};
+
+use test_case::test_case;
+
+use once_cell::sync::Lazy;
+use std::sync::Mutex;
+
+struct TestContext {
+    ledger_dir: PathBuf,
+    pparams_dir: PathBuf,
+}
+
+fn get_test_context() -> Result<TestContext, Box<dyn std::error::Error>> {
+    let working_dir = env::current_dir()?;
+    Ok(TestContext {
+        ledger_dir: working_dir.join(PathBuf::from("../../ledger.db")),
+        pparams_dir: working_dir.join(PathBuf::from("../../cardano-blueprint/src/ledger/conformance-test-vectors/eras/conway/impl/dump/pparams-by-hash/")),
+    })
+}
+
+static TEST_MUTEX: Lazy<Mutex<()>> = Lazy::new(Mutex::default);
+
+include!("generated_ledger_conformance_test_cases.incl");
+
+fn evaluate_vector(snapshot: &str) -> Result<(), Box<dyn std::error::Error>> {
+    let _shared = TEST_MUTEX.lock()?;
+    let network = NetworkName::Testnet(1);
+    let era_history = network.into();
+    match get_test_context() {
+        Ok(tc) => {
+            let vector_file = fs::read(snapshot)?;
+            let record: NESVector = minicbor::decode(&vector_file)?;
+            let () = import_vector(record, &tc.ledger_dir, era_history, &tc.pparams_dir)?;
+            Ok(())
+        }
+        Err(e) => {
+            tracing::warn!(
+                "skipping vector evaluation because some env vars are missing: {}",
+                e
+            );
+            Ok(())
+        }
+    }
+}
+
+fn peek_epoch(nes: &AnyCbor) -> Result<u64, Box<dyn std::error::Error>> {
+    let mut d = cbor::decode::Decoder::new(nes.raw_bytes());
+    let _ = d.array()?;
+    Ok(d.u64()?)
+}
+
+#[derive(cbor::Decode)]
+#[allow(dead_code)]
+struct NESVector {
+    #[n(0)]
+    initial_nes: AnyCbor,
+    #[n(1)]
+    final_nes: AnyCbor,
+    #[n(2)]
+    transactions: Vec<AnyCbor>,
+    #[n(3)]
+    test_state: String,
+}
+
+fn import_vector(
+    record: NESVector,
+    ledger_dir: &PathBuf,
+    era_history: &EraHistory,
+    pparams_dir: &PathBuf,
+) -> Result<(), Box<dyn std::error::Error>> {
+    if record.initial_nes[0] == 0x80 {
+        return Ok(());
+    }
+    let epoch = peek_epoch(&record.initial_nes)?;
+    let point = amaru_kernel::Point::try_from(
+        format!(
+            "{}.0000000000000000000000000000000000000000000000000000000000000000",
+            epoch * 86400
+        )
+        .as_str(),
+    )?;
+
+    fs::create_dir_all(ledger_dir)?;
+    let db = RocksDB::empty(ledger_dir)?;
+
+    let network = NetworkName::Testnet(1);
```

Comment:
Seems like this should come from the arguments, since the import data are already network-dependent.